DashSet does not implement serde

DashMap implements serde, but why not for DashSet?

`error[E0277]: the trait bound `dashmap::DashSet<u8>: helloworld::_::_serde::Deserialize<'_>` is not satisfied`

--- a/src/serde.rs
+++ b/src/serde.rs
@@ -1,9 +1,9 @@
-use crate::DashMap;
+use crate::{DashMap, DashSet};
 use core::fmt;
 use core::hash::Hash;
-use serde::de::{Deserialize, MapAccess, Visitor};
+use serde::de::{Deserialize, MapAccess, SeqAccess, Visitor};
 use serde::export::PhantomData;
-use serde::ser::{Serialize, SerializeMap, Serializer};
+use serde::ser::{Serialize, SerializeMap, SerializeSeq, Serializer};
 use serde::Deserializer;
 
 pub struct DashMapVisitor<K, V> {
@@ -75,5 +75,74 @@ where
         }
 
         map.end()
+    }
+}
+
+pub struct DashSetVisitor<K> {
+    marker: PhantomData<fn() -> DashSet<K>>,
+}
+
+impl<K> DashSetVisitor<K>
+where
+    K: Eq + Hash,
+{
+    fn new() -> Self {
+        DashSetVisitor {
+            marker: PhantomData,
+        }
+    }
+}
+
+impl<'de, K> Visitor<'de> for DashSetVisitor<K>
+where
+    K: Deserialize<'de> + Eq + Hash,
+{
+    type Value = DashSet<K>;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("a DashSet")
+    }
+
+    fn visit_seq<M>(self, mut access: M) -> Result<Self::Value, M::Error>
+    where
+        M: SeqAccess<'de>,
+    {
+        let map = DashSet::with_capacity(access.size_hint().unwrap_or(0));
+
+        while let Some(key) = access.next_element()? {
+            map.insert(key);
+        }
+
+        Ok(map)
+    }
+}
+
+impl<'de, K> Deserialize<'de> for DashSet<K>
+where
+    K: Deserialize<'de> + Eq + Hash,
+{
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_seq(DashSetVisitor::<K>::new())
+    }
+}
+
+impl<K> Serialize for DashSet<K>
+where
+    K: Serialize + Eq + Hash,
+{
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut seq = serializer.serialize_seq(Some(self.len()))?;
+
+        for ref_multi in self.iter() {
+            seq.serialize_element(ref_multi.key())?;
+        }
+
+        seq.end()
     }
 }

